luaui/Widgets/export_defs.lua

```diff
@@ -0,0 +1,60 @@
+function widget:GetInfo()
+	return {
+		name = "Unitdefs JSON Export",
+		desc = "Exports UnitDefs into JSON files.",
+		author = "Wereii",
+		date = "June 2023",
+		license = "GNU GPL, v2 or later",
+		layer = 99999,
+		enabled = false
+	}
+end
+
+local export_folder_path = "json_export"
+
+local function TableToFile(tbl, filename)
+	local file, err = io.open(filename, "w")
+	if not file then
+		Spring.Echo("Error opening file: " .. err)
+		return
+	end
+	file:write(Json.encode(tbl))
+	file:close()
+end
+
+local function ExportDefs()
+	if not VFS.FileExists(export_folder_path) then
+		Spring.CreateDir(export_folder_path)
+	end
+
+	for id, unitDef in pairs(UnitDefs) do
+		Spring.Echo(string.format("Exporting unitdef: %s", unitDef.name))
+		local tbl = {}
+
+		tbl["id"] = unitDef.id
```

Comment:
why do this explicitly, isn't it contained in the table? i'd actually recommend doing it the other way around and removing it manually after iterating over all fields, since the value is assigned arbitrarily and should never be used directly.

Reply:
hmm I thought the id was not being set for whatever reason, checking it without the line and it works alright  
removed

Reply:
> should never be used directly.

It's still referenced in for example `buildOptions`, would have to do two passes to resolve all the ids.